dtfs-central-api/src/services/tfm/deal-cancellation.service.ts

```diff
@@ -100,6 +114,29 @@ export class DealCancellationService {
       dealType,
     });
 
-    return response;
+    await Promise.all(
+      riskExpiredFacilities.map(({ _id: facilityId }) =>
+        PortalFacilityService.updateStatus({ facilityId, status: FACILITY_STATUS.RISK_EXPIRED, dealType, auditDetails }),
+      ),
+    );
+
+    return this.getTfmDealCancellationResponse({ cancelledDeal, riskExpiredFacilities });
+  }
+
+  /**
+   * Maps values returned by repository to return deal cancellation DTO
+   *
+   * @param repositoryResponse - The response from the deal cancellation repository method
+   * @returns Tfm deal cancellation response DTO
+   */
+  public static getTfmDealCancellationResponse({ cancelledDeal, riskExpiredFacilities }: { cancelledDeal: TfmDeal; riskExpiredFacilities: TfmFacility[] }) {
+    const cancelledDealUkefId = getUkefDealId(cancelledDeal.dealSnapshot) as string;
+
+    return {
+      cancelledDealUkefId,
+      riskExpiredFacilityUkefIds: riskExpiredFacilities
+        .map(({ facilitySnapshot }) => facilitySnapshot.ukefFacilityId)
+        .filter((facilityId) => facilityId !== null),
```

Comment:
At this stage, are we ever going to have facilities that do not have a `ukefFacilityId`?

Reply:
We shouldn't, but otherwise typescript isn't happy with this. I guess theres a couple alternatives
- throw an error in the `map` if they're null (note that this means no email will be sent, but the deal will have been cancelled)
- assert with an `as string` that these are never  `null`

Reply:
String assertion sounds nicest to me open to other approaches